cni-plugin/pkg/dataplane/linux/dataplane_linux.go

```diff
@@ -362,10 +373,18 @@ func SetupRoutes(hostNlHandle *netlink.Handle, hostVeth netlink.Link, result *cn
 					}
 				}
 
-				// Search all routes and report the conflict, search the name of the iface
-				routes, err = hostNlHandle.RouteList(nil, netlink.FAMILY_ALL)
-				if err != nil {
-					return fmt.Errorf("error listing routes")
+				retries = 3
+				for {
+					// Search all routes and report the conflict, search the name of the iface
+					routes, err = hostNlHandle.RouteList(nil, netlink.FAMILY_ALL)
+					if err != nil {
+						if errors.Is(err, unix.EINTR) && retries > 0 {
+							retries--
+							continue
+						}
+						return fmt.Errorf("error listing routes")
```

Comment:
Probably want to keep the original error wrapped up in this one when we return it